src/java/main/org/apache/zookeeper/server/ServerStats.java

```diff
@@ -148,9 +174,46 @@ synchronized public void resetRequestCounters(){
         packetsReceived = 0;
         packetsSent = 0;
     }
+    synchronized public void resetNumRequestsAboveThresholdTime() {
+        numRequestsAboveThresholdTime = 0;
+    }
     synchronized public void reset() {
         resetLatency();
         resetRequestCounters();
+        resetNumRequestsAboveThresholdTime();
+    }
+
+    public void checkLatency(final ZooKeeperServer zks, Request request) {
+        long requestLatency = Time.currentElapsedTime() - request.createTime;
+        boolean enabledAndAboveThreshold = (requestWarnThresholdMs == 0) ||
+                (requestWarnThresholdMs > -1 && requestLatency > requestWarnThresholdMs);
```

Comment:
This could be simplified: `boolean enabledAndAboveThreshold = requestWarnThresholdMs != -1 && requestLatency > requestWarnThresholdMs`